challenges/ai-ml/challenge-1.md

```diff
@@ -0,0 +1,250 @@
+  sufficient.
+* Avoid sending us draft design documents. It is difficult to evaluate which
+  parts are draft and which parts are complete. Instead we encourage asking
+  questions in Slack and sharing a design document that is ready to be reviewed.
+
+Be sure to cover the following in your design:
+
+* Your approach to solving the challenge problem.
+* CLI user experience.
+
+Once the design document has been approved by two reviewers, move on to the
+implementation.
+
+## Implementation
+
+Split your code submission into roughly 2-3 Pull Requests to give the team an
+opportunity to review your code and provide feedback. Feel free to merge each
+PR after you have two approvals.
+
+Our team will do their best to provide a high quality review of the submitted
+Pull Requests in a reasonable time frame. You are spending your time on this,
+we are going to contribute our time too.
+
+After the final submission, the interview team will assemble and vote using a
+"+1, -2" anonymous voting system: +1 is submitted whenever a team member
+accepts the submission, -2 otherwise.
+
+In case of a positive result, we will connect you to our HR and recruiting
+teams, who will work out the details and present an offer.
+
+In case of a negative score result, the hiring manager will contact you and share a
+list of the key observations from the team that affected the result.
+
+### Tools
+
+This task should be implemented in Python, Go or Rust and should work on 64-bit
+Linux or macOS machines.
+
+### Testing
+
+Key components of the challenge should have tests that cover the happy and
+unhappy scenarios. Do not try to achieve 100% test coverage as that will take
+too long.
+
+### Dependencies
+
+Please write as much of your own code as possible. Avoid relying on third
+party dependencies for key components of the challenge where possible. Do not
+rely on any shell scripts or external binaries.
+
+Prefer the standard library and whichever CLI library you are most familiar
+with. If there is a dependency that you want to use but are unsure about please
+ask first.
+
+# Requirements
+
+## Level 4
+
+Use the public dataset of CloudTrail logs from flaws.cloud to implement a CLI
+tool that is able to detect a cloud account reconnaissance activity using
+anomaly detection and/or machine learning techniques.
+
+https://summitroute.com/blog/2020/10/09/public_dataset_of_cloudtrail_logs_from_flaws_cloud/
+https://summitroute.com/downloads/flaws_cloudtrail_logs.tar
+
+Reconnaisance activity may include enumeration of AWS services and resources,
+permissions probing and brute-force attempts, unusual API call frequencies.
+
+After analyzing the logs, the tool should output the following information:
+
+- Whether the suspicious activity was detected and the confidence level.
+- The suspected identity.
+- The time frame of the suspected recon activity.
+- Key indicators (example logs/API calls).
+
+Include a README documentation with the tool.
```

Comment:
Should we ask them to implement the best approach for this or any approach is valid?
I am wondering because we say to keep the challenge simple but in reality we want candidates to apply the best possible approach